File: sql/verifica.java
/**
 * 
 * @author Claudio Invernizzi <cinvernizzi@dsgestion.site>
 * @name: verifica
 * @date 14/10/2025
 * @Projecto: Presupuesto
 * @Copyright DsGestion <dsgestion.site>
 * @Licence: GPL
 * Clase que verifica la existencia de las distintas bases de 
 * datos y en todo caso las crea
 * 
 */

// definición del paquete
package sql;

// definición de la clase
public class verifica {

    /**
     * @author Claudio Invernizzi <cinvernizzi@dsgestion.site>
     * Constructor de la clase, abrimos la conexión y verifica
     * que existan las distintas tablas
     * 
     * Tenemos que tener en cuenta el orden porque sqlite si bien 
     * soporta las claves foráneas, solo las podemos definir en 
     * el momento de su creación, así debemos crear primero los 
     * diccionarios 
     */
    public verifica(){

        // verificamos la tabla de clientes que no tiene 
        // referencias externas
        new Clientes();

        // verificamos los datos propios que no tiene 
        // referencias externas
        new Personal();

        // verificamos la tabla de secciones que no tiene 
        // referencias externas
        new Secciones();

        // la tabla de proyectos que tiene como referencia
        // externa los clientes
        new Proyectos();

        // la tabla de presupuestos que tiene como referencia
        // la tabla de proyectos
        new Presupuesto();

        // verificamos la tabla de actividades que tiene como 
        // referencia la tabla de proyectos, la tabla de 
        // secciones 
        new Actividades();
        
        // ahora creamos las vistas
        new Vistas();

    }

}

File: sql/Verifica.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

"""
    Nombre: Verifica.py
    Autor: Lic. Claudio Invernizzi
    Fecha: 16/10/2025
    E-Mail: cinvernizzi@dsgestion.site
    Proyecto: Presupuesto
    Comentarios: Clase que verifica la existencia de las distintas
                 tablas

"""

# importamos las librerías
from sql.Actividades import Actividades
from sql.Clientes import Clientes
from sql.Personal import Personal
from sql.Secciones import Secciones
from sql.Proyectos import Proyectos
from sql.Presupuesto import Presupuesto
from sql.Vistas import Vistas

class Verifica:
    """

        @author Claudio Invernizzi <cinvernizzi@dsgestion.site>
    
        Definición de la clase

    """

    def __init__(self):
        """
        
            @author Claudio Invernizzi <cinvernizzi@dsgestion.site>

            Constructor de la clase que simplemente instancia las 
            distintas verificacions de la base de datos local

        """

        # verificamos los clientes
        Clientes()

        # la tabla de datos personales
        Personal()

        # insertamos las secciones
        Secciones()

        # los datos de los proyectos
        Proyectos()

        # los datos de los presupuestos
        Presupuesto()

        # verificamos las actividades
        Actividades()

        # ahora creamos las vistas
        Vistas()
